### ISSUE TITLE
[C#] Implement new Concept Exercise: time

### ISSUE THREAD

**Issue by mikedamay** [exercism/v3#1644]
This issue describes how to implement the `time` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how to use `time` in C#.

## Learning objectives

- Know how to use `time` in C#.
- Know how to get the current time.
- Know how to perform arithmetic on times.
- Know the difference between local time and UTC.

## Out of scope

- [Noda Time](https://nodatime.org/)

## Concepts

- `time`
- `timezone`

## Prerequisites

- `datetime`

Other prerequisites will depend on the story.

### Hints
This [article][time-overview] is a good overview of time and timezone

### After
This [article][time-overview] is a good overview of time and timezone

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[time-overview]: https://docs.microsoft.com/en-us/dotnet/standard/datetime/
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer

**Comment by mikedamay** [comment]
Please have a look at my ideas in the [instructions](https://github.com/mikedamay/v3/blob/csharp/time/languages/csharp/exercises/concept/time/.docs/instructions.md) and [code](https://github.com/mikedamay/v3/blob/csharp/time/languages/csharp/exercises/concept/time/.meta/Example.cs).

Timezones give us a cross-site problem.  I have chosen to provide the ids in the instructions as this is a simple approach.  Alternatives are to detect the platform.

**Comment by ErikSchierboom** [comment]
@mikedamay I really like it!

**Comment by mikedamay** [comment]
closed by #1805
